### PR TITLE
Add setup.sh script to install and update the bot

### PR DESCRIPTION
**Pull request by glonlas**
## Summary
This PR adds a script to Install/Update/Reset the bot on MacOS and Linux. It was created to help users to install the bot.

## Script features
The script `setup.sh` can
- Install the bot from scratch
- Update dependencies on an existing bot
- Reset a branch and virtual env
- Configure config.json

## Script usage
```
$ ./setup.sh
usage:
	-i,--install    Install freqtrade from scratch
	-u,--update     Command git pull to update.
	-r,--reset      Hard reset your develop/master branch.
	-c,--config     Easy config generator (Will override your existing file).
	-p,--plot       Install dependencies for Plotting scripts.
```

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+#encoding=utf8
+freqtradepath=$(pwd)
+
+function go_to_folder () {
```
Comment (shusso) [inline setup.sh]: do we really need a function for this?

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+echo "You can now use the bot by executing 'source .env/bin/activate; python3 freqtrade/main.py'."
+}
+
+function plot () {
+echo "
+-----------------------------------------
+Install dependencies for Plotting scripts
+-----------------------------------------
+"
+pip install plotly
```
Comment (shusso) [inline setup.sh]: you could also use `pip install plotly --upgrade` if the user has a old installation

**Comment by glonlas** [comment]
Thank you @shusso. This is fixed

**Comment by glonlas** [comment]
@shusso could you tell me if you have other comments?

**Pull request merged**